core/src/main/java/org/elasticsearch/rest/action/cat/RestThreadPoolAction.java

```diff
@@ -260,9 +150,31 @@ private Table buildTable(RestRequest req, ClusterStateResponse state, NodesInfoR
                     }
                 }
             }
-            for (String pool : SUPPORTED_NAMES) {
-                ThreadPoolStats.Stats poolStats = poolThreadStats.get(pool);
-                ThreadPool.Info poolInfo = poolThreadInfo.get(pool);
+            for (Map.Entry<String, ThreadPoolStats.Stats> entry : poolThreadStats.entrySet()) {
+
+                boolean matches = false;
+                for (final String threadPool : threadPools) {
+                    matches = entry.getKey().matches(threadPool);
```

Comment:
Sorry, I meant `o.e.common.regex.Regex#simpleMatch` specifically. It is more like shell globing than a regex and we use it all over the public API. So, yeah, I mean behavior consistency.

Also I think `simpleMatch` is less prone to stupid backtracking issues than arbitrary regular expressions are. Which is a small plus in any API.